lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceReject.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+   Copyright 2020 The Magma Authors.
+   This source code is licensed under the BSD-style license found in the
+   LICENSE file in the root directory of this source tree.
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#include <sstream>
+#include "M5GServiceReject.h"
+#include "M5GCommonDefs.h"
+
+namespace magma5g {
+ServiceRejectMsg::ServiceRejectMsg(){};
+ServiceRejectMsg::~ServiceRejectMsg(){};
+
+// Decoding Service Reject Message and its IEs
+int ServiceRejectMsg::DecodeServiceRejectMsg(
```

Comment:
write unit tests for encoding/decoding. i.e., encode to buffer, check success, decode the buffer, check success, check if decoded matches to the original.

Reply:
Done